nifty-ssl/src/main/java/com/facebook/nifty/ssl/NettyTcNativeLoader.java

```diff
@@ -0,0 +1,223 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import com.google.common.base.Throwables;
+import org.apache.tomcat.jni.Library;
+import org.apache.tomcat.jni.SSL;
+import org.jboss.netty.handler.ssl.OpenSsl;
+import org.jboss.netty.util.internal.NativeLibraryLoader;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Adds support for the Netty-tcnative uber jar. The current version of netty nifty runs with does not support the
+ * uber jar. This should be removed when we move to new version of netty.
+ *
+ * Most of this code is taken from
+ * https://github.com/netty/netty/blob/379ad2c02ed0c0ae9f94e4081e3f910ece6380b7/handler/src/main/java/io/netty/handler/ssl/OpenSsl.java
+ */
+public class NettyTcNativeLoader {
+
+    private static final String LINUX = "linux";
+    private static final String UNKNOWN = "unknown";
+
+    private static final Throwable UNAVAILABILITY_CAUSE;
+
+    static {
+        Throwable cause = null;
+        try {
+            loadTcNative();
+        } catch (Throwable t) {
+            cause = t;
+        }
+        UNAVAILABILITY_CAUSE = cause;
+    }
+
+    public static void ensureAvailable() {
+        if (UNAVAILABILITY_CAUSE != null) {
+            throw (Error) new UnsatisfiedLinkError(
+                    "failed to load the required native library").initCause(UNAVAILABILITY_CAUSE);
+        }
+    }
+
+    private static void loadTcNative() {
+        String os = normalizeOs(System.getProperty("os.name", ""));
+        String arch = normalizeArch(System.getProperty("os.arch", ""));
+
+        Set<String> libNames = new LinkedHashSet<String>(3);
+        // First, try loading the platform-specific library. Platform-specific
+        // libraries will be available if using a tcnative uber jar.
+        libNames.add("netty-tcnative-" + os + '-' + arch);
+        if (LINUX.equalsIgnoreCase(os)) {
+            // Fedora SSL lib so naming (libssl.so.10 vs libssl.so.1.0.0)..
+            libNames.add("netty-tcnative-" + os + '-' + arch + "-fedora");
+        }
+        // finally the default library.
+        libNames.add("netty-tcnative");
+
+        boolean loaded = false;
+        Throwable lastException = null;
+        for (String libName : libNames) {
+            try {
+                NativeLibraryLoader.load(libName, SSL.class.getClassLoader());
+                loaded = true;
+                break;
+            } catch (Throwable t) {
```

Comment:
Formatting: the `}` should be on it's own line:
```java
try {
    ...
}
catch {
    ...
}
```
See https://github.com/airlift/codestyle